### DISCUSSION TITLE
another noob quastion

### DISCUSSION THREAD

**Discussion by astrobudka** [aaronwmorris/indi-allsky#528] [Q&A]
after ./setup.sh he ask me in 32 lines witch cam will i use. 1. altair 2.apogee 3.asi..... i have a imx462 camera, but  non usb connected  but this slim cable. Witch of indi driver  i should choice ?

**Comment by aaronwmorris** [comment]
Do you have a link to the product you purchased?

**Reply by astrobudka** [reply]
yes, siure

https://www.aliexpress.com/item/1005004608219711.html?spm=a2g0o

**Reply by aaronwmorris** [reply]
I will work on getting support added today.

**Reply by aaronwmorris** [reply]
I have added some additional code to help with this.  I am not sure what is necessary to get imx462 support in libcamera.  It may require libcamera itself to be compiled from source.

When you run setup.sh, select libcamera and then libcamera_imx462.

**Reply by astrobudka** [reply]
now i have two instalations. First the old one after /setup.sh have 23 cameras to select but no libcamera on list. 
The new intalation make a week ago have only two cameras to select after /setup.sh 1-indi 2-libcamera_imx477

Dont know what to do now. 3rd instalation ?

**Reply by aaronwmorris** [reply]
You need to pull down the latest code using `git pull origin`.  After this, the selection above for libcamera_imx477 will be replaced with libcamera and then you can select additional libcamera variants.

In reality, you can select any of the libcamera options just to see if the camera will work.  If you select the wrong libcamera camera, the worst problem you will have is the wrong debayering algorithm will be applied.

**Reply by astrobudka** [reply]
so i make 3rd installetion. ububtu pi server (pi 3+ B) after some not reconize stuff with i figure out...  ./setup.sh i get 2 cameras, 1. indi 2. libcam..... i select 2end then imx462.... there  are no rebuilt packeges for thise distribution. Please run ./misc/build_indi.sh witch i did.... 
some long time later...  (18800sec)... another try ./setup.sh... select europe, warsow... and now: Installing build dependencies in progres.  keep fingers cross :)

**Reply by aaronwmorris** [reply]
It has been my experience that libcamera is only available on Raspbian 11 (64-bit works best).

**Comment by bluthen** [comment]
My guess is raspberry pi camera. They sell IMX462 modules that are libcamera compat, like this one. There has been a couple I was thinking of trying. Some good candidates for an allsky camera use.
https://www.arducam.com/product/arducam-for-raspberry-pi-ultra-low-light-camera-1080p-hd-wide-angle-pivariety-camera-module-based-on-1-2-7inch-2mp-starvis-sensor-imx462-compatible-with-raspberry-pi-isp-and-gstreamer-plugin/

**Comment by astrobudka** [comment]
with imx462 i gave up. I ordered imx291 with usb port.
https://aliexpress.com/item/1005001998432509.html

During the installation, I selected the INDI option and the sony_ccd sensor.

VLC in rasberypi captures the image without any problems, allsky says it can't see the camera.

In config I set INDI, imx290... doesn't help. There is no image to display.

Allsky's status is unknown. I remember that with the zwo camera the status was up when it worked and down when I changed the parameters by restarting the server.

Instructions:
systemctl --user start indiserver
systemctl --user start indi-allsky
executes without problem, but when I used ./allsky.py run reported critical problems.

Any tips?

Sorry for taking your time, but I care about allsky and I have beginner problems.

**Reply by aaronwmorris** [reply]
No worries.  For the imx291, you would probably have to select INDI and indi_webcam_ccd.

The camera you have selected is primarily a video streaming camera, it is not very capable for taking snapshots for timelapses.  It will likely be very limited.

I would recommend something like the imx378 to ensure compatibility with libcamera and ind-allsky.  https://www.aliexpress.us/item/3256803721407425.html

**Reply by astrobudka** [reply]
It works !!!   :) 

Few times :(
No i have #500error on web browser. 

Internal Server Error
The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.

From local chromium and external pc. Already reboot pi many times, no efect :(

**Reply by aaronwmorris** [reply]
Check the `/var/log/indi-allsky/webapp-indi-allsky.log` file for the exception.

I would try clearing your cookies or using a private browser session and see if that helps.

**Reply by astrobudka** [reply]
All log since i boot RPi:

Mar 26 20:36:28 raspberrypi gunicorn.indi_allsky.wsgi.error: [723] WORKER TIMEOUT (pid:887)
Mar 26 20:36:28 raspberrypi [INFO] MainProcess glogging.info() #264: Worker exiting (pid: 887)
Mar 26 20:36:29 raspberrypi gunicorn.indi_allsky.wsgi.error: [723] Worker with pid 887 was terminated due to signal 9
Mar 26 20:36:29 raspberrypi gunicorn.indi_allsky.wsgi.error: [1630] Booting worker with pid: 1630
Mar 26 20:38:06 raspberrypi [ERROR] MainProcess app.log_exception() #1744: Exception on /indi-allsky/ [GET]#012Traceback (most recent call last):#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1964, in _exec_single_context#012 self.dialect.do_execute(#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 748, in do_execute#012 cursor.execute(statement, parameters)#012sqlite3.DatabaseError: database disk image is malformed#012#012The above exception was the direct cause of the following exception:#012#012Traceback (most recent call last):#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/flask/app.py", line 2528, in wsgi_app#012 response = self.full_dispatch_request()#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/flask/app.py", line 1825, in full_dispatch_request#012 rv = self.handle_user_exception(e)#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/flask/app.py", line 1823, in full_dispatch_request#012 rv = self.dispatch_request()#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/flask/app.py", line 1799, in dispatch_request#012 return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)#012 File "/home/allsky/indi-allsky/indi_allsky/flask/misc.py", line 22, in decorated_view#012 return current_app.ensure_sync(func)(*args, **kwargs)#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/flask/views.py", line 104, in view#012 self = view.view_class( # type: ignore[attr-defined]#012 File "/home/allsky/indi-allsky/indi_allsky/flask/base_views.py", line 101, in __init__#012 super(TemplateView, self).__init__(**kwargs)#012 File "/home/allsky/indi-allsky/indi_allsky/flask/base_views.py", line 41, in __init__#012 self._indi_allsky_config_obj = IndiAllSkyConfig()#012 File "/home/allsky/indi-allsky/indi_allsky/config.py", line 274, in __init__#012 config_entry = self._getConfigEntry()#012 File "/home/allsky/indi-allsky/indi_allsky/config.py", line 321, in _getConfigEntry#012 config_entry = IndiAllSkyDbConfigTable.query\#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/orm/query.py", line 2806, in one#012 return self._iter().one() # type: ignore#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/orm/query.py", line 2855, in _iter#012 result: Union[ScalarResult[_T], Result[_T]] = self.session.execute(#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 2229, in execute#012 return self._execute_internal(#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 2124, in _execute_internal#012 result: Result[Any] = compile_state_cls.orm_execute_statement(#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/orm/context.py", line 253, in orm_execute_statement#012 result = conn.execute(#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1414, in execute#012 return meth(#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 486, in _execute_on_connection#012 return connection._execute_clauseelement(#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1638, in _execute_clauseelement#012 ret = self._execute_context(#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1842, in _execute_context#012 return self._exec_single_context(#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1983, in _exec_single_context#012 self._handle_dbapi_exception(#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 2326, in _handle_dbapi_exception#012 raise sqlalchemy_exception.with_traceback(exc_info[2]) from e#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1964, in _exec_single_context#012 self.dialect.do_execute(#012 File "/home/allsky/indi-allsky/virtualenv/indi-allsky/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 748, in do_execute#012 cursor.execute(statement, parameters)#012sqlalchemy.exc.DatabaseError: (sqlite3.DatabaseError) database disk image is malformed#012[SQL: SELECT config.id AS config_id, config."createDate" AS "config_createDate", config.level AS config_level, config.encrypted AS config_encrypted, config.note AS config_note, config.data AS config_data, config.user_id AS config_user_id #012FROM config ORDER BY config."createDate" DESC#012 LIMIT ? OFFSET ?]#012[parameters: (1, 0)]#012(Background on this error at: https://sqlalche.me/e/20/4xp6)

**Reply by aaronwmorris** [reply]
Looks like the database is corrupted.  You can follow these instructions to rebuild it.

https://github.com/aaronwmorris/indi-allsky/wiki/Database-migrations#rebuilding-database

**Reply by astrobudka** [reply]
Works!

**Reply by astrobudka** [reply]
Sorry for taking so much yours time ,
i set 30 sec max sub time and gain for a moon mode 30 and night mode 45. I make darknes at room and wait some time. After taking a look at description on fotos, the sub time is 30 sec but gain 1. AllSky is at  moon mode. 
![gain](https://user-images.githubusercontent.com/123208963/228056938-685cf10a-35b0-406a-8256-02911b32f721.jpg)

**Reply by aaronwmorris** [reply]
Great!
